packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -268,15 +273,38 @@ export class SummaryGenerator {
 		);
 
 		let summaryData: SubmitSummaryResult | undefined;
+
+		// Called when summarization fails at any stage. Validates that the right params are passed when a particular
+		// stage failed. Logs Summarize_cancel event.
 		const fail = (
```

Comment:
While you're here, consider switching to a params object since there are so many optional things. Then you name everything at the callsite and don't have to pass undefined with a /* paramName */ comment for stuff that's not relevant

Reply:
I reverted most of the changes here. Please take a look.